security/MemberLoginForm.php

```diff
@@ -213,8 +213,8 @@ protected function logInUserAndRedirect($data) {
 		}
 
 		// Absolute redirection URLs may cause spoofing
-		if(!empty($_REQUEST['BackURL'])) {
-			$url = $_REQUEST['BackURL'];
+		if(!empty($data['BackURL'])) {
+			$url = $data['BackURL'];
```

Comment:
If this form is posted, does $data include both $_POST and $_GET vars in the same way that $_REQUEST does?

Reply:
MemberLoginForm does this

```
// Reduce attack surface by enforcing POST requests
		$this->setFormMethod('POST', true);
```
		
So $data won't include an $_GET vars.

doLogin() is passed the HTTPRequest as a third parameter though. You want to update to use this instead?

Reply:
Sorry, I just realised that on postback BackURL is a hidden field not a querystring. Probably safe then?